tools/copy_bom/src/util.rs

```diff
@@ -495,3 +531,51 @@ fn deal_with_output(output: Output, error_number: i32) {
         std::process::exit(error_number);
     }
 }
+
+/// Add missing library to a global hashset
+fn add_missing_library(missing_library: String) {
+    MISSING_LIBRARIES
+        .lock()
+        .expect("Acquire lock should not fail")
+        .insert(missing_library);
+}
+
+/// Check whether missing library exists in image directory
+fn check_missing_library(missing_library: &str, image_dir: &str) -> bool {
+    let mut command = Command::new("find");
```

Comment:
Can we create a copied_file list to maintain those files already added into the image_dir? Then we could search the missing files in the copied_list.
The difference is: that the copied_list is consistent; it would not change if the image_dir is not clean at the beginning. It could help the user generate the correct and self-consistent bom file.

Reply:
`copy_bom` can only track files defined in **copyfiles** now, ***NOT*** including files in **copydirs**. The process to copy directory is controlled by rsync only. So, if users pack libraries in a directory, `copy_bom` cannot track such libraries.

Reply:
> `copy_bom` can only track files defined in **copyfiles** now, _**NOT**_ including files in **copydirs**. The process to copy directory is controlled by rsync only. So, if users pack libraries in a directory, `copy_bom` cannot track such libraries.

Get it, good point. As mitigation, can we print a warning message if the target folder  (i.e. image folder) is not empty at the beginning?

Reply:
Yes, I will fix it to give a warning message.